tests/unit/conftest.py

```
@@ -0,0 +1,88 @@
+    Patches the sync and async executors' ``execute`` to record ``self.statement()``
+    and return an empty result, so ``schema.execute_sync`` runs fully DB-free.
+
+    Args:
+        monkeypatch: pytest fixture used to patch the executors' ``execute`` methods.
+
+    Returns:
+        A list appended with one statement per executor execution.
+    """
+    captured: list[Select[Any]] = []
+
+    def _execute(self: SyncQueryExecutor[Any], session: Any) -> MagicMock:  # noqa: ARG001
```

Ruff (0.15.21)

[warning] Dynamically typed expressions (typing.Any) are disallowed in `session`

(ANN401)